### DISCUSSION TITLE
Possible Bug/Need Help: Macro conditions are excecuting all at once without the correct timing

### DISCUSSION THREAD

**Discussion by eriksenchee** [WarmUpTill/SceneSwitcher#537] [General]
Yo!

I just recently updated the plugin in order to use it for OBS 28.0 and I noticed that I bumped into some issues with creating macros. I mainly use macros to initate a bunch of camera changes at specific timeframes but in this verision it doesnt seem to work as it used to. 

For context: I have conditions set where if i press a HOTKEY, it should switch to TOP CAM scene and after 5 secs -> then BACK CAM. But instead it just does it all at once.

I tried messing around with different condition, as well as reinstalling the plugin but I can't seem to figure out what is possibly missing.

Below I attached a preview of what happens. 

p.s this plugin is amazing and perfect my respects

[https://streamable.com/3omzaw](url)

**Comment by eriksenchee** [comment]
I just tried reverting back to the old version for both the plugin and OBS 27.xx and I noticed that the same issue is happening. So it could potentially be a missed setting or something is bugged with my files.

**Comment by nacalai** [comment]
I see the problem. The "with a duration of x seconds" means that it will transition to the specified scene using the specified transition for x amount of seconds. So if you choose fade, it will fade over for 5 seconds. But here it's set to cut. So, it will just cut instantly. 

You have to add a a wait-action and put in the delay you want to actually have delays between actions :)

**Reply by eriksenchee** [reply]
THANK U, this clears it up. Appreciate you alot.

**Comment by WarmUpTill** [comment]
Correct, the duration field in the "Switch scene" action is specific to the transition type you have selected.
For the "Cut" transition specifying a duration value does not have any effect, as this transition happens instantly by design.

What you are looking to do is something like this, as already suggested by @nacalai :)

![Wait](https://user-images.githubusercontent.com/19472752/190868509-80e01233-7195-47a0-9e66-9900574da55e.PNG)

I will see if I can maybe hide the duration field if the selected transition type does not support specifying a transition duration.

**Reply by eriksenchee** [reply]
Ahh I see, I must of been using a dated version of the plugin because I sware thats how I originally set it up. Thank you so much and for all the work you do :)

**Reply by eriksenchee** [reply]
Damn, After putting the wait conditions, the issue is still there. 

The swap to Camera 4 (TOP CAM) is when I first press the hotkey, then it stays on it longer than 5s and doesn't change. It's supposed to hopellly switch to BACK CAM.

Here is another video to show: [https://streamable.com/ybz889](url)

**Reply by WarmUpTill** [reply]
The settings _seem_ correct and do work for me.
Have you configured anything else in the plugin? (Can you maybe export the settings to a file using the Option on the General tab)
What type of transition are you using?
Does unchecking "Wait until transition to target scene is complete" make a difference?

**Reply by eriksenchee** [reply]
It finally worked! I just restarted my OBS Studio and the plugin works as expected.

Thank you for helping me as always :))))))
